.github/actions/secure-project-checkout-go/action.yml

```diff
@@ -22,8 +22,8 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: Checkout the repository with user SHA
-      uses: slsa-framework/slsa-github-generator/.github/actions/secure-checkout@ea457948e6a989e9818304dcb4f15fb4ce6765d6
+    - name: Checkout the repository with user ref
+      uses: ./__BUILDER_CHECKOUT_DIR__/.github/actions/secure-project-checkout
```

Comment:
doesn't this need a secure builder dir checkout before use?
same below for node. or maybe I'm missing something

Reply:
maybe it propogates since it's an action and run in the same environment? if so, maybe this action should validate that BUILDER_CHECKOUT_DIR is populated?

Reply:
you're correct, it assumes the builder repo is checkout'ed. The checkout is done in the main re-usable workflow by calling `uses: slsa-framework/slsa-github-generator/.github/actions/secure-builder-checkout`.

The actions run in the same context as the job, so it's present.

What do you mean by "maybe this action should validate that BUILDER_CHECKOUT_DIR is populated"? The action will fail if the directory is not present, so the validation is implicit. Do you have something else in mind?

Btw, BUILDER_CHECKOUT_DIR is not an env variable, because `uses:` does not accept them.

Let me know if this clarifies or still confusing.

Reply:
> The action will fail if the directory is not present, so the validation is implicit.

Only for a clear error message: maybe can you add a comment to the action.yml saying it requires that the repository is populated? Just in case it hits an error when creating a new builder